sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -140,35 +140,68 @@ void *malloc_device(size_t Size, const device &Dev, const context &Ctxt) {
   return detail::usm::alignedAlloc(0, Size, Ctxt, Dev, alloc::device);
 }
 
+void *malloc_device(size_t Size, const queue &Q) {
+  return detail::usm::alignedAlloc(0, Size, Q.get_context(), Q.get_device(),
```

Comment:
Suggest redirecting to malloca_device(size_t, device, context) to avoid code duplication.

Reply:
Good call.